### PR TITLE
Extract Logger into its own package

### PR DESCRIPTION
**Pull request by webtaculars**
# ↪️ Pull Request
Extract logger code into a separate package inside the monorepo

closes  [#2134]

## ✔️ PR Todo

- [x] Added/updated unit tests for this change
- [x] Filled out test instructions (In case there aren't any unit tests)
- [x] Included links to related issues/PRs

### CODE REVIEW

File: packages/core/parcel-bundler/src/Asset.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ const md5 = require('./utils/md5');
 const isURL = require('./utils/is-url');
 const config = require('./utils/config');
 const syncPromise = require('./utils/syncPromise');
-const logger = require('./Logger');
+const logger = require('@parcel/Logger');
```
Comment (devongovett) [inline packages/core/parcel-bundler/src/Asset.js]: should be lower case: `@parcel/logger`

File: packages/core/parcel-bundler/test/integration/babel-plugin-autoinstall/package.json [hunk 2]
```diff
@@ -1,3 +1,6 @@
 {
-  "name": "babel-plugin-autoinstall"
+  "name": "babel-plugin-autoinstall",
+  "devDependencies": {
+    "@babel/core": "^7.1.2"
```
Comment (devongovett) [inline packages/core/parcel-bundler/test/integration/babel-plugin-autoinstall/package.json]: this shouldn't change - part of the test is asserting that this is added.

**Comment by devongovett** [comment]
Awesome, thanks! Can you rebase with master?

**Comment by devongovett** [comment]
I removed the original emoji and prettyError files from utils and exposed them as part of the `@parcel/logger` package instead.

**Comment by interglobalmedia** [comment]
I have parcel-bundler 1.11.0 installed locally (don't want to install globally) and using npx as a result. Everything was fine in development, but can't build my assets for production. Am getting the following in Terminal console:
```
npm run build                                                      ⏎ ✖ ✹

> html5-drag-and-drop@0.0.1 build /Users/mariacam/Development/html5-drag-and-drop
> npx parcel build index.js -d dist

⚠️  Cannot find module 'parcel-bundler/src/Logger'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:603:15)
    at Function.Module._load (internal/modules/cjs/loader.js:529:25)
    at Module.require (internal/modules/cjs/loader.js:658:17)
    at require (/Users/mariacam/Development/html5-drag-and-drop/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
    at Object.<anonymous> (/Users/mariacam/Development/html5-drag-and-drop/node_modules/parcel-plugin-eslint/index.js:1:78)
    at Module._compile (/Users/mariacam/Development/html5-drag-and-drop/node_modules/v8-compile-cache/v8-compile-cache.js:178:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:733:10)
    at Module.load (internal/modules/cjs/loader.js:620:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:560:12)
    at Function.Module._load (internal/modules/cjs/loader.js:552:3)
🚨  No entries found.
    at Bundler.bundle (/Users/mariacam/Development/html5-drag-and-drop/node_modules/parcel-bundler/src/Bundler.js:261:17)
```
Does parcel-bundler not work with npx???

**Comment by DeMoorJasper** [comment]
@interglobalmedia are you using any plugins?

**Comment by interglobalmedia** [comment]
I have since removed parcel, but yes I was. If I remember correctly, the parcel-plugin-bundle-manifest, parcel-plugin-eslint, and parcel-plugin-static-files-copy.

**Comment by DeMoorJasper** [comment]
@interglobalmedia then it’s definitely a plugin issue. Plugins shouldn’t be using internal parcel code (including the logger)

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by interglobalmedia** [comment]
I'm not very familiar with parcel. What does that mean? And then why are there plugins to extend parcel? Thanks!

**Comment by DeMoorJasper** [comment]
It means that parcel/src/Logger shouldn’t be imported directly but by using Bundler.logger instead as internal code is not protected by semver

The plugin architecture wasn’t perfect in parcel 1. It should be easier to extend functionality once Parcel 2 gets released.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request merged**